sdk/helper/ldaputil/client.go

```diff
@@ -120,6 +120,10 @@ func (c *Client) GetUserBindDN(cfg *ConfigEntry, conn Connection, username strin
 		}
 
 		filter := fmt.Sprintf("(%s=%s)", cfg.UserAttr, ldap.EscapeFilter(username))
+		if cfg.UPNDomain != "" {
```

Comment:
When I run the tests in `client_test.go`, I see that none of them hit this line, which might be part of why we were hitting this problem in the first place. Do you think it would be possible to add test coverage to this using a Docker container like [this one](https://hub.docker.com/r/osixia/openldap/)? Might be a nice way to start increasing the stability of this package.

Reply:
I find that's true with the other sections of the code, as well.

Reply:
Thanks, @tyrannosaurus-becks. I added a test which tests the UPNDomain functionality.